docs/rfc-1-staking-contract.adoc

```diff
@@ -0,0 +1,381 @@
+
+=== Recovering a stake
+
+==== `recoverStake(address operator) external`
+
+Sets the undelegating delegation state as recovered once the undelegation period
+passes. For liquid T delegations, transfers tokens back to the original owner.
+Can be called by anyone.
+
+=== Keeping information in sync
+
+==== `notifyKeepStakeDiscrepancy(address operator)`
+
+Notifies about the discrepancy between legacy KEEP stake and stake amount in T
+staking contract. Can be called by anyone, notifier receives a reward.
+
+Optionally: reward withdrawal can be split into a separate function to protect
+against MEV frontrunners. 
+
+==== `notifyNuStakeDiscrepancy(address operator)`
+
+Notifies about the discrepancy between legacy NU stake and stake amount in T
+staking contract. Can be called by anyone, notifier receives a reward.
+
+Optionally: reward withdrawal can be split into a separate function to protect
+against MEV frontrunners. 
+
+==== `setStakeDiscrepancyPenalty(uint256 penalty, unit256 rewardMultiplier) external onlyGovernance`
+
+Sets the penalty amount for stake discrepancy and reward multiplier for
+reporting it. The penalty is seized from the operator account, and 5% of the
+penalty, scaled by the multiplier, is given to the notifier. The rest of the
+tokens are burned. Can only be called by the governance. See `seize` function.
+
+=== Slashing a stake
+
+==== `slash(uint256 amount, address[] memory operators) external onlyAuthorizedApplication`
+
+Slashes the provided amount from the stake of every operator in the array.
+Can only be called by an authorized application.
+
+==== `seize(uint256 amount, uint256 rewardMultipier, address notifier, address[] memory operators) external onlyAuthorizedApplication`
+
+Seize the provided amount from the stake of every operator in the array.
+The misbehavior notifier is rewarded with 5% of the total seized amount scaled
+by the reward adjustment parameter. The rest of the tokens are burned. Can only
+be called by an authorized application.
+
+=== Auxiliary functions
+
+==== `eligibleStake(address operator, address application) external view returns (uint256)`
+
+Returns the eligible stake amount of the operator for the application.
+An eligible stake is a stake that is not currently undelegating and the given
+application had to be approved for use by the operator's authorizer. If the
+stake is undelegation or the application was not approved, returns 0.
+
+The difference between eligible and active stake is that the active stake does
+not make the operator eligible for new work selection but it may still finish
+earlier work it is assigned to before the stake is recovered.
+
+==== `activeStake(address operator, address application) external view returns (uint256)`
+
+Returns the active stake amount of the operator for the application. An active
+stake is a stake that may be in the process of undelegation but has not been
+recovered yet. The given application had to be approved for use by the
```

Comment:
`has not finished undelegating yet`; stake that hasn't been recovered but has passed the undelegation period should not be active